assets/prototype/domain/eventEditor/datesList/dateCard/DateCard.js

```diff
@@ -31,75 +33,75 @@ const DateCard = ({ eventId, id }) => {
 	const onFieldUpdate = useUpdateDateMutation({ id });
 
 	const startDate = date.start ? new Date(date.start * 1000) : PLUS_ONE_MONTH;
-	// const endDate = date.end ?
-	// 	new Date( date.end * 1000 ) :
-	// 	PLUS_TWO_MONTHS;
 
 	return (
-		<Card elevation={Elevation.ONE} style={cardStyle}>
-			<div>
-				<div style={idStyle}>{date.datetimeId}</div>
-				<H4>
-					<EditableText
-						placeholder='Edit title...'
-						defaultValue={date.name}
-						onCancel={(value) => console.log(value)}
-						onConfirm={(name) => onFieldUpdate({ name })}
-						minWidth={'320px'}
-						selectAllOnFocus
-					/>
-				</H4>
-			</div>
-			<div>
-				<H6>
-					<EditableText
-						placeholder='Edit description...'
-						defaultValue={date.description}
-						onCancel={(value) => console.log(value)}
-						onConfirm={(description) => onFieldUpdate({ description })}
-						minWidth={'320px'}
-						multiline={true}
-						maxLines={4}
-						selectAllOnFocus
-					/>
-				</H6>
-			</div>
-			<DateRangeInput endDate={date.endDate} onFieldUpdate={onFieldUpdate} startDate={date.startDate} />
-			<div>
-				<b>{`${date.startDate} ${date.startTime}`}</b>
-				<Popover lazy>
-					<Button icon='calendar' style={btnStyle} minimal />
-					<DatePicker
-						defaultValue={startDate}
-						formatDate={(jsDate) => jsDate.toString()}
-						onChange={(jsDate, isUserChange) => {
-							if (jsDate && isUserChange) {
-								date.start = jsDate.getTime() / 1000;
-								date.startDate = jsDate.toDateString();
-								date.startTime = jsDate.toLocaleTimeString();
-								console.log(
-									'%c update state date: ',
-									'color: #F2F500; font-size:14px;',
-									'%c > date.start: ',
-									'color: #99c043;',
-									new Date(date.start * 1000)
-								);
-							}
-						}}
-						timePrecision={TimePrecision.MINUTE}
-						timePickerProps={{
-							showArrowButtons: true,
-							useAmPm: true,
-						}}
-						minDate={A_LONG_TIME_AGO}
-						maxDate={PLUS_TEN_YEARS}
-						highlightCurrentDay
-						showActionsBar
-					/>
-				</Popover>
-			</div>
-			<DeleteDateButton eventId={eventId} id={date.id} />
-		</Card>
+		<DateTimeProvider id={id}>
+			<Card elevation={Elevation.ONE} style={cardStyle}>
+				<EditDate />
```

Comment:
I don't need to pass any prop here, this component is living in the context of `DateCard`
and is smart enough to pick what it needs.